raceSystem/webUI.py

```diff
@@ -0,0 +1,194 @@
+from flask import Flask, render_template, request, jsonify
+from raceSystem.userInterface import UserInterface
+
+
+class WebIU(UserInterface):
+    def __init__(self) -> None:
+        self._app = Flask(__name__)
+        self._startClientConnected = False
+        self._finishClientConnected = False
+        self._raceStatus = ''
+        self._raceResult = ['', '']
+
+    def run(self):
+        """Start the user interface"""
+        self.setupFlaskRouting()
+        self._app.run(debug=True, use_reloader=False)
+
+    def setupFlaskRouting(self):
+        self.homePage()
+        self.formRaceControl()
+        self.formNames()
+        self.settingsPage()
+        self.statusRequest()
+
+    def homePage(self):
+        @self._app.route("/", methods=['GET', 'POST'])
+        def index():
+            name1 = self.getNameCallback(0)
+            name2 = self.getNameCallback(1)
+            return render_template(
+                "home.html",
+                name_1=name1 + " ",
+                name_2=name2 + " ")
+
+    def formRaceControl(self):
+        @self._app.route("/raceControl", methods=['POST'])
+        def raceControl():
+            if request.method == 'POST':
+                if request.form.get('btnStart') == 'Start':
+                    self.startRaceCallback()
+                elif request.form.get('btnStop') == 'Stop':
+                    self.stopRaceCallback()
+            return '', 204
+
+    def formNames(self):
+        @self._app.route("/setNames", methods=['POST', 'GET'])
+        def setNames():
+            if request.method == 'POST':
+                self.setNameCallback(0, request.form.get('Naam_1'))
+                self.setNameCallback(1, request.form.get('Naam_2'))
+
+            return self.homePage()
+
+    def statusRequest(self):
+        @self._app.route('/get-status')
+        def getStatus():
+            return jsonify(
+                xStartClientConnected=self._startClientConnected,
+                xFinishClientConnected=self._finishClientConnected,
+                sRaceStatus=self._raceStatus,
+                sRaceResult_1=self._raceResult[0],
+                sRaceResult_2=self._raceResult[1])
+
+    def settingsPage(self):
```

Comment:
Changing settings doesn't work anymore if flask is started with a defines host id. Somehow it tries to save settings to 127.0.0.1:5000

Reply:
Fixed.